catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/QueryResponseImpl.java

```diff
@@ -109,19 +111,52 @@ public QueryResponseImpl(
       boolean closeResultQueue,
       long hits,
       Map<String, Serializable> properties) {
+    this(request, results, closeResultQueue, hits, properties, null);
+  }
+
+  /**
+   * Instantiates a new {@code QueryResponseImpl} with: a {@link QueryRequest}, a {@link List} of
+   * {@link Result}s from the {@link QueryRequest}, an indicator of whether to close this {@code
+   * QueryResponseImpl}'s {@link #queue} of {@link Result}s, the number of distinct {@link Result}s
+   * which this {@code QueryResponseImpl} contains, a {@link Map} which contains the keys and values
+   * of this {@code QueryResponseImpl}'s properties, and the {@link Set} of {@link
+   * ProcessingDetails} from the execution of the {@link QueryRequest}
+   *
+   * @param request the {@link QueryRequest}
+   * @param results the {@link List} of {@link Result}s
+   * @param shouldCloseResultQueue the indicator of whether to close the {@link #queue}
+   * @param hits the number of distinct {@link Result}s
+   * @param properties the {@link Map} of this {@code QueryResponseImpl}'s properties' keys to their
+   *     values
+   * @param processingDetails the {@link Set} of {@link ProcessingDetails} of the {@link
+   *     QueryRequest}'s execution
+   */
+  public QueryResponseImpl(
+      QueryRequest request,
+      List<Result> results,
+      boolean shouldCloseResultQueue,
+      long hits,
+      Map<String, Serializable> properties,
+      Set<ProcessingDetails> processingDetails) {
```

Comment:
✏️ Consider consolidating constructor parameters into one or two domain objects to reduce parameter clutter

Reply:
👎 I personally would not do that. The classes would incur more overhead than they resolve.

Reply:
If we're using Sonar as part of our [Code Quality Policy](https://jenkins.phx.connexta.com/service/jenkins/job/DDF/job/master/) in DDF this will break the build
https://rules.sonarsource.com/java/RSPEC-107

Reply:
There's a lot about Sonar that is good. But I don't think we fix every finding that gets reported. We're only at six arguments, that's not terrible. If we need to add more, then it might be time to reconsider.